portal-api/src/v1/users/sign-in-link.service.js

```diff
@@ -17,37 +17,47 @@ class SignInLinkService {
   async createAndEmailSignInLink(user) {
     const { _id: userId, email: userEmail, firstname: userFirstName, surname: userLastName } = user;
 
-    const signInCode = this.#createSignInCode();
-    await this.#saveSignInCodeHashAndSalt({ userId, signInCode });
+    const signInToken = this.#createSignInToken();
+    await this.#saveSignInTokenHashAndSalt({ userId, signInToken });
 
     return this.#sendSignInLinkEmail({
-      signInLink: `${PORTAL_UI_URL}/login/sign-in-link?t=${signInCode}`,
+      signInLink: `${PORTAL_UI_URL}/login/sign-in-link?t=${signInToken}`,
       userEmail,
       userFirstName,
       userLastName,
     });
   }
 
-  #createSignInCode() {
+  async isValidSignInToken({ userId, signInToken }) {
+    const user = await this.#userRepository.findById(userId);
+    if (!user.hash || !user.salt) {
+      throw new Error('User does not have a valid sign in token.');
+    }
```

Comment:
Why do we have this check?

Reply:
We actually never have checked that this exists in the database prior to this step. On the off chance that we've somehow got here and failed to save this to the database (currently error handled), we should be prepared to handle it here too do you think? I didn't want to base the implimentation around other functions error handling